File: ios/Anemone SDK/model/tenant/BookingType.swift
//
//  BookingType.swift
//  Anemone SDK
//
//  Created by Angel Luis Garcia on 26/09/2017.
//  Copyright © 2017 Syltek Solutions S.L. All rights reserved.
//

import Foundation

public enum BookingType: String {
    case `public` = "PUBLIC", restricted = "RESTRICTED", `private` = "PRIVATE"

    static func fromRawValue(_ rawValue: String) -> BookingType? {
        BookingType(rawValue: rawValue.uppercased())
    }
}

extension BookingType: CustomStringConvertible {
    public var description: String {
        rawValue
    }
}

File: android/app/src/main/java/com/playtomic/test/anemone_sdk/model/tenant/BookingType.kt
package com.anemonesdk.model.tenant

import com.playtomic.foundation.model.CustomStringConvertible

/**
 * Created by agarcia on 26/09/2017.
 */
enum class BookingType constructor(val rawValue: String) : CustomStringConvertible {
    PUBLIC("PUBLIC"),
    RESTRICTED("RESTRICTED"),
    PRIVATE("PRIVATE");

    override val description get() = rawValue

    companion object {
        fun fromRawValue(value: String): BookingType? = entries.firstOrNull { it.rawValue == value.uppercase() }
    }
}
